gammapy/time/plot.py

```diff
@@ -2,31 +2,126 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 
-__all__ = ['plot_time_difference_distribution',
+from ..datasets import fetch_fermi_catalog
+
+__all__ = ['plot_light_curve',
            ]
 
 
-def plot_time_difference_distribution(time, ax=None):
-    """Plot event time difference distribution.
+def plot_light_curve(name_3FGL, fermi_met_start = 2.3933e8, fermi_met_end = 3.65467550e8):
+    """Plot flux as a function of time for a fermi 3FGL object.
 
     Parameters
     ----------
-    time : `~astropy.time.Time`
-        Event times (must be sorted)
-    ax : `~matplotlib.axes.Axes` or None
-        Axes
-
-    Returns
-    -------
-    ax : `~matplotlib.axes.Axes`
-        Axes
+    name_3FGL : `string`
+        The 3FGL catalog name of the object to plot
+    fermi_met_start : `int`
+        Start time of the light curve in Fermi MET
+    fermi_met_end : `int`
+        End time of the light curve in Fermi MET
+
+    Usage
+    gammapy.time.plot_light_curve('3FGL J0349.9-2102', 2.1e8, 3.2e8)
     """
+    import numpy as np
     import matplotlib.pyplot as plt
+    import math
+
+    try:
+        fermi_met_start = int(fermi_met_start)
+    except:
+        print("Fermi_met_start not a valid numerical input.")
+        return
+
+    try:
+        fermi_met_end = int(fermi_met_end)
+    except:
+        print("Fermi_met_end not a valid numerical input.")
+        return
+
+    # As far as I know light curves were only included in 3FGL, so we must use this catalog.
+    fermi_cat = fetch_fermi_catalog('3FGL')
+
+    try:
+        catalog_index = np.where(fermi_cat[1].data['Source_Name'] == name_3FGL)[0][0]
+    except IndexError:
```

Comment:
You should remove all of the try ... except in this function.

At least in library code like you're writing here it's important to raise Exceptions or Warnings when something goes wrong. Just printing an error message means that callers don't have a chance to catch the issue if they e.g. produce 1000s or plots in a pipeline and one source name is wrong.

Of course sometimes it can make sense to catch errors and re-raise more specific errors, or to catch warnings and silence them completely because it's expected behaviour in that case, but it's rare.

Plus we can't write Gammapy so that we check inputs all the time.
It would literally mean writing ~ 2x as much code.
See some comments I made here: https://gammapy.readthedocs.org/en/latest/development/index.html#what-checks-and-conversions-should-i-do-for-inputs